applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp

```diff
@@ -0,0 +1,272 @@
+    CheckProperty(rMaterialProperties, GEO_FRICTION_ANGLE);
+    CheckProperty(rMaterialProperties, GEO_DILATANCY_ANGLE);
+    CheckProperty(rMaterialProperties, GEO_TENSILE_STRENGTH);
+    CheckProperty(rMaterialProperties, YOUNG_MODULUS);
+    CheckProperty(rMaterialProperties, POISSON_RATIO);
+    return 0;
+}
+
+void MohrCoulombWithTensionCutOff::CheckProperty(const Properties& rMaterialProperties,
+                                                 const Kratos::Variable<double>& rVariable) const
+{
+    constexpr auto min_value = 0.0;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(rVariable) || rMaterialProperties[rVariable] < min_value)
+        << rVariable.Name()
+        << " is not defined or has an invalid value for property: " << rMaterialProperties.Id()
+        << std::endl;
+}
+
+ConstitutiveLaw::StressMeasure MohrCoulombWithTensionCutOff::GetStressMeasure()
+{
+    return ConstitutiveLaw::StressMeasure_Cauchy;
+}
+
+SizeType MohrCoulombWithTensionCutOff::GetStrainSize() const
+{
+    return mpConstitutiveDimension->GetStrainSize();
+}
+
+ConstitutiveLaw::StrainMeasure MohrCoulombWithTensionCutOff::GetStrainMeasure()
+{
+    return ConstitutiveLaw::StrainMeasure_Infinitesimal;
+}
+
+bool MohrCoulombWithTensionCutOff::IsIncremental() { return true; }
+
+bool MohrCoulombWithTensionCutOff::RequiresInitializeMaterialResponse() { return true; }
+
+void MohrCoulombWithTensionCutOff::InitializeMaterial(const Properties& rMaterialProperties,
+                                                      const Geometry<Node>&,
+                                                      const Vector&)
+{
+    const auto friction_angle_in_rad = MathUtils<>::DegreesToRadians(rMaterialProperties[GEO_FRICTION_ANGLE]);
+    const auto cohesion = rMaterialProperties[GEO_COHESION];
+    const auto dilatancy_angle = MathUtils<>::DegreesToRadians(rMaterialProperties[GEO_DILATANCY_ANGLE]);
+    const auto tension_cutoff = rMaterialProperties[GEO_TENSILE_STRENGTH];
+
+    mCoulombYieldSurface = CoulombYieldSurface(friction_angle_in_rad, cohesion, dilatancy_angle);
+    mTensionCutOff       = TensionCutoff(tension_cutoff);
+}
+
+void MohrCoulombWithTensionCutOff::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters& rParameters)
+{
+    const Properties& r_prop         = rParameters.GetMaterialProperties();
+    const auto friction_angle_in_rad = MathUtils<>::DegreesToRadians(r_prop[GEO_FRICTION_ANGLE]);
+    const auto cohesion              = r_prop[GEO_COHESION];
+    const auto dilatancy_angle       = MathUtils<>::DegreesToRadians(r_prop[GEO_DILATANCY_ANGLE]);
+    const auto tension_cutoff        = r_prop[GEO_TENSILE_STRENGTH];
+    const auto young_modulus         = r_prop[YOUNG_MODULUS];
+    const auto poisson_ratio         = r_prop[POISSON_RATIO];
+
+    Vector trail_stress_vector;
+    Vector strain_vector = rParameters.GetStrainVector();
+
+    this->CalculateTrialStressVector(strain_vector, trail_stress_vector, young_modulus, poisson_ratio);
```

Comment:
```suggestion
    this->CalculateTrialStressVector(rParameters.GetStrainVector(), trail_stress_vector, r_prop[YOUNG_MODULUS], r_prop[POISSON_RATIO]);
```

In this way, we also no longer need the (constant) variables `young_modulus` and `poisson_ratio`.